src/cpp/src/visual_language/cdpruner/cdpruner.cpp

```diff
@@ -0,0 +1,627 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cdpruner.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstring>
+#include <future>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <thread>
+
+#include "openvino/openvino.hpp"
+#include "utils.hpp"
+
```

Comment:
Suggest to add a simple algorithm flowchart here or header file.
Somethin like:

prompts feature --> 
vision feature      -->     RelevanceCalculator --> get redundant vision feature --> update original vision feature.